impl/src/planning/subplanners.rs

```diff
@@ -236,25 +213,63 @@ impl<'planner> WorkspaceSubplanner<'planner> {
       .iter()
       .sorted_by_key(|n| &n.id)
       .filter_map(|node| self.create_crate_context(node, &self.crate_catalog))
-      .collect::<Result<Vec<CrateContext>>>()
+      .collect::<Result<Vec<_>>>()?;
+
+    let root_ctxs = contexts
+      .iter()
+      .filter_map(|(ctx, is_workspace)| match is_workspace {
+        true => Some(ctx.default_deps.aliased_dependencies.clone()),
+        false => None,
+      })
+      .flatten()
+      .collect::<BTreeSet<_>>();
+
+    let contexts = contexts.into_iter().map(|(ctx, _)| ctx).collect_vec();
+    let aliases = self.produce_workspace_aliases(root_ctxs, &contexts);
+
+    Ok((contexts, aliases))
+  }
+
+  fn produce_workspace_aliases(
+    &self,
+    root_dep_aliases: BTreeSet<DependencyAlias>,
+    all_packages: &[CrateContext],
+  ) -> Vec<DependencyAlias> {
+    let renames = root_dep_aliases
+      .iter()
+      .map(|rename| (&rename.target, &rename.alias))
+      .collect::<HashMap<_, _>>();
+
+    all_packages
+      .iter()
+      .filter(|to_alias| to_alias.lib_target_name.is_some())
+      .flat_map(|to_alias| {
+        let pkg_name = util::sanitize_ident(&to_alias.pkg_name);
```

Comment:
I feel like the sanitize is worth keeping but wanted to share https://github.com/google/cargo-raze/commit/9d74090f28cf7d546414a0591205efa5219c9bf2 here in case that's relevant. I think `sanatize_indent` might need to be updated

Reply:
Oops I thought I factored that in when I rebased so good catch.